Contents/mods/Skill Recovery Journal/media/lua/client/Skill Recovery Journal XP.lua

```diff
@@ -117,16 +117,16 @@ local function catchJavaAddXP(player, perksType, XP)
     ---checking if it's 'not false' instead of 'true' because I want older saves before this sandbox option to get what they expect to occur
     if SandboxVars.SkillRecoveryJournal.RecoverProfessionAndTraitsBonuses == false then
```

Comment:
Also, based on the comment above it, it seems this statment may need the `~=` operator instead of `==`. I'm not sure exactly what this statement does, though. Best for you to review it, @Chuckleberry-Finn .

Reply:
This is an issue when adding new sandboxOptions - previous saves will have a 'nil' value rather than the default value. So I made it operate to respect older saves without the sandbox option set.

Reply:
I'm actually not sure why this is this way... it should be `if not x == false`.

Reply:
I did go ahead and update the operator to `~=` in the next commit after first commenting this, so double check the logic still looks right to you. I've spotted multiple of your comments for the backwards compatible logic, so maybe double check those as well.

This mod is really nice work, by the way.

Reply:
Thank you, and yeah I re-re-read the ~= false again. I think it's cause it's a returned value, so anything not false returns true. Nil and true are not false so it returns true.